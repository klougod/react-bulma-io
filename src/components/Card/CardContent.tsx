import React from 'react'
import PropTypes from 'prop-types'

export const CardContent = ({ className, children }: any) => (
  <header className={`card-content ${className}`}>{children}</header>
)

CardContent.defaultProps = {
  className: ''
}

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default CardContent
