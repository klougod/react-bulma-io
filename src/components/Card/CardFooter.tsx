import React, { ReactNode } from 'react'

interface CardFooterProps {
  className?: string,
  children?: ReactNode
}

export const CardFooter = ({ className, children }: CardFooterProps) => (
  <header className={`card-footer ${className}`}>{children}</header>
)

CardFooter.defaultProps = {
  className: ''
}

export default CardFooter
