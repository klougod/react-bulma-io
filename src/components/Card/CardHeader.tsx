import React from 'react'
import PropTypes from 'prop-types'

export const CardHeader = ({ className, children }: any) => (
  <header className={`card-header ${className}`}>{children}</header>
)

CardHeader.defaultProps = {
  className: ''
}

CardHeader.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default CardHeader
