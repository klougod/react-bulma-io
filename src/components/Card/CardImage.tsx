import React from 'react'
import PropTypes from 'prop-types'
import Figure from '../Figure'

export const CardImage = ({ className, alt, src, children }: any) => (
  <div className='card-image'>
    {src ? <Figure src={src} alt={alt} className={className} /> : { children }}
  </div>
)

CardImage.defaultProps = {
  className: '',
  alt: '',
  src: ''
}

CardImage.propTypes = {
  className: PropTypes.string,
  alt: PropTypes.string,
  src: PropTypes.string,
  children: PropTypes.node
}

export default CardImage
