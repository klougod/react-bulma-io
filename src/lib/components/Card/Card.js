import React, { Children } from 'react'
import PropTypes from 'prop-types'

export const Card = ({ className, children }) => (
  <div className={`card ${className}`}>
    { Children.toArray(children).filter(c => c.type.name === "CardHeader") }
    { Children.toArray(children).filter(c => c.type.name === "CardImage") }
    { Children.toArray(children).filter(c => c.type.name === "CardContent") }
    { Children.toArray(children).filter(c => c.type.name === "CardFooter") }
  </div>
)

Card.defaultProps = {
  className: ''
}

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Card