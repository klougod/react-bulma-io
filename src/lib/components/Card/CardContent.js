import React from 'react'
import PropTypes from 'prop-types'

export const CardContent = ({ className, children }) => (
  <header className={`card-content ${className}`}>
    { children }
  </header>
)

CardContent.defaultProps = {
  className: ''
}

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardContent