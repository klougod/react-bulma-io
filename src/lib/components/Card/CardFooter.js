import React from 'react'
import PropTypes from 'prop-types'

export const CardFooter = ({ className, children }) => (
  <header className={`card-footer ${className}`}>
    { children }
  </header>
)

CardFooter.defaultProps = {
  className: ''
}

CardFooter.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardFooter