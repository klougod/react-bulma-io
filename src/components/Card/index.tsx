import React, { Children, ReactElement } from 'react'
import PropTypes from 'prop-types'

import CardHeader from './CardHeader'
import CardImage from './CardImage'
import CardContent from './CardContent'
import CardFooter from './CardFooter'

const cardHeaderType = (<CardHeader />).type
const cardImageType = (<CardImage />).type
const cardContentType = (<CardContent />).type
const cardFooterType = (<CardFooter />).type

export const Card = ({ className, children }: any) => (
  <div className={`card ${className}`}>
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === cardHeaderType
    )}
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === cardImageType
    )}
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === cardContentType
    )}
    {Children.toArray(children).filter(
      (c: ReactElement) => c.type === cardFooterType
    )}
  </div>
)

Card.defaultProps = {
  className: ''
}

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export { CardHeader, CardImage, CardContent, CardFooter }
export default Card
