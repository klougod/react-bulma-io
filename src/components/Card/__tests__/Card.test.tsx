import React from 'react'
import Enzyme from 'enzyme'

import Card from '../'
import CardHeader from '../CardHeader'
import CardContent from '../CardContent'
import CardImage from '../CardImage'
import CardFooter from '../CardFooter'

describe('Card component test', function () {
  it('should render without crash', function () {
    const cardMock = Enzyme.shallow(<Card />)
    expect(cardMock).toMatchSnapshot()
  })

  it('should render without crash with correct children', function () {
    const cardMockProps = { className: 'cardClass' }
    const cardMock = Enzyme.shallow(
      <Card {...cardMockProps}>
        <CardHeader />
        <CardContent />
        <CardImage />
        <CardFooter />
      </Card>
    )
    expect(cardMock).toMatchSnapshot()
  })
})
