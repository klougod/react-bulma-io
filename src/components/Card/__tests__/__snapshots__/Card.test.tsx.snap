// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test should render without crash 1`] = `
<div
  className="card "
/>
`;

exports[`Card component test should render without crash with correct children 1`] = `
<div
  className="card cardClass"
>
  <Component
    className=""
    key=".0"
  />
  <Component
    alt=""
    className=""
    key=".2"
    src=""
  />
  <Component
    className=""
    key=".1"
  />
  <Component
    className=""
    key=".3"
  />
</div>
`;
